### PR TITLE
Ft survey migrations

### LINKED ISSUE #22: Create Survey Migrations
**Action required**
Create the migrations needed for the survey form

### PR DESCRIPTION
**Pull request by onabright**
## Description
I have created 5 models and migrations for capturing survey data  as follows:
**Models**

1. Location
2. Question
3. ResponseOption
4. SurveyResponse
5.SurveyResponse2

**Migrations**
**locations:** Contains information about the area where the survey is being conducted (typically a slum area)
**questions:** Contains information about all the questions that will appear on the survey
**response_options:** Contains information about the expected responses from each question. Typically these appear as select option values
**survey_responses:** Contains submitted data from each individual survey conducted
**survey_responses_2:** Contains submitted data from each individual survey conducted. An alternative approach to survey_response

Fixes #22 

## How Has This Been Tested?
No specific tests where done

## Screenshots
For some strange reason I get the error "Something went really wong, and we can't process that file" when I try to upload screenshots

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added necessary inline code documentation
- [ ] I have added tests that prove my fix is effective and that this feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .idea/Team-269-Backend.iml
Comment (geneowak) [inline .idea/Team-269-Backend.iml]: I think you should add the .idea folder to the gitginore since not every one uses an IDE...
Take a look at this (https://coderwall.com/p/gfjgwg/remove-ide-files-from-your-git-repository-but-keeping-in-your-local-directory) on how you can easily achieve that....

File: app/SurveyResponse.php
Comment (geneowak) [inline app/SurveyResponse.php]: Is this for the user who took the survey or for the respondents?

File: app/SurveyResponse2.php
Comment (geneowak) [inline app/SurveyResponse2.php]: Hmm... Do we need this? I think we may need to do a refactor

File: app/User.php
Comment (geneowak) [inline app/User.php]: Same thing here, are we referring to the users taking the survey or those participating in it?
If the latter then I think this may not be useful since most surveys don't collect information that identifies the respondents

File: config/database.php
Comment (geneowak) [inline config/database.php]: Let us leave this as pgsql since it is what Heroku depends on, you can override it in your own .env file.

File: database/migrations/2020_05_29_092823_create_locations_table.php
Comment (geneowak) [inline database/migrations/2020_05_29_092823_create_locations_table.php]: How about having these in their own table since there will be duplication?
We could have the locations table with the details of the different locations and then we have the the slums table with the slum names and their location id.

File: database/migrations/2020_05_29_095139_create_survey_responses_table.php
Comment (geneowak) [inline database/migrations/2020_05_29_095139_create_survey_responses_table.php]: you can merge this like so
`$table->foreignId('question_id')->constrained()->onDelete('cascade');`
you would then do away with the code on line 22.

you could take a look at this (https://laravel.com/docs/7.x/migrations#foreign-key-constraints) for more info

File: database/migrations/2020_05_29_102439_create_survey_response2s_table.php
Comment (geneowak) [inline database/migrations/2020_05_29_102439_create_survey_response2s_table.php]: I don't think we should hard code these because we will have a hard time if we are told to expand the range of the data we collect. Let us make them dynamic, we can create seeders to add the predefined responses. 

I see these could easily go in the questions table, we could pair on it and see the best way to go about it...

**Review by geneowak — CHANGES_REQUESTED** [review]
Great work. 
A few things to correct.
1. Remove the IDE files form version control.
2. There are a few other changes I've suggested, let me know if you would like us to pair on them...

PS: I've left some references for more info.

**Review by MutegekiHenry — COMMENTED** [review]
Good work @onabright just work on those changes and we are good

File: app/SurveyResponse.php
Comment (onabright) [reply app/SurveyResponse.php]: User in this context is the person conducting the survey (the enumerator). Every survey response that is submitted by an enumerator (user) should be tagged to that enumerator.

File: .idea/Team-269-Backend.iml
Comment (onabright) [reply .idea/Team-269-Backend.iml]: Yeah! I noticed after I had made the PR that I should have added the .idea to the .gitignore. Let me fix that! Thanks

File: app/SurveyResponse2.php
Comment (onabright) [reply app/SurveyResponse2.php]: That will depend on the approach we take. I added this so that we can review and see whether it works for our context.

File: app/User.php
Comment (onabright) [reply app/User.php]: The former. The user (enumerator) conducting the survey. The respondent does not interact with the system. The enumerator asks the questions and fills out the responses provided by the respondent, and then submits the response. We are not collecting any information that identifies the respondent. But we would like to know who administered the survey (the user/enumerator)

File: config/database.php
Comment (onabright) [reply config/database.php]: Sorry I thought I had added config/database.php to the .gitignore. I use mysql in my dev environment so I changed this. I should have changed it back to pgsql. 
Let me fix this.

File: database/migrations/2020_05_29_092823_create_locations_table.php
Comment (onabright) [reply database/migrations/2020_05_29_092823_create_locations_table.php]: OK.  This is noted.

File: database/migrations/2020_05_29_095139_create_survey_responses_table.php
Comment (onabright) [reply database/migrations/2020_05_29_095139_create_survey_responses_table.php]: Noted.

File: database/migrations/2020_05_29_102439_create_survey_response2s_table.php
Comment (onabright) [reply database/migrations/2020_05_29_102439_create_survey_response2s_table.php]: OK. noted.

File: app/SurveyResponse.php
Comment (geneowak) [reply app/SurveyResponse.php]: I think this may not be necessary since a response belongs to a survey, we should pull the user who submitted from the survey itself. We should have the user who collects/submits the survey data as the user responsible for all the survey data otherwise we will have to track down whoever collected each response and that can get tedious.
In a real scenario, there's always one overall person responsible for a given survey. If we want to also note the people who participated in collecting the data, then we could have something like a survey_team table to hold that info.

**Review by geneowak — DISMISSED** [review]
I think we are good now, make some checks on your side and merge as soon as we have another approval.

**Review by MutegekiHenry — APPROVED** [review]

**Review by onabright — COMMENTED** [review]
Changes look good. @geneowak please merge.

**Pull request merged**
